dev/Kozani/KozaniRemoteManager/ConnectionManager.h

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation and Contributors.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "KozaniRemoteManagerTraceLogging.h"
+#include "KozaniDvcServer.h"
+
+namespace Microsoft::Kozani::KozaniRemoteManager
+{
+    class ConnectionManager
+    {
+    public:
+        friend class KozaniDvcServer;
+
+        ConnectionManager() = default;
+        ~ConnectionManager() noexcept;
+
+        void Connect(PCSTR connectionId);
+        void ReportDvcServerError(HRESULT hr);
+
+    private:
+        void WaitForFailedDvcServerDeletionThread();
+        UINT64 GetNewActivityId();
+        void ProcessAppTermination(AppActivationInfo* appActivationInfo);
+
+    private:
+        bool m_closing{};
+        bool m_isModuleObjectCountAdded{};
+
+        // Do not use wil::srwlock as it does not support recursive locking, which may be needed in this object.
```

Comment:
I haven't done full analysis of the threading pattern but there is a possibility that it can get into recursive locking. So the wording is accurate. Better be conservative and use recursive_mutex than finding deadlock later.